### DISCUSSION TITLE
Ability to intentionally choose Model Attributes to fill the forms.

### DISCUSSION THREAD

**Discussion by LogicSatinn** [filamentphp/filament#7856] [Ideas]
Since Livewire v3 doesn't support some custom property types, it can cause issues for those who have casted their Model properties to unsupported property types. Bearing in mind the existence of Livewire Synths to automatically add support for these custom properties, sometimes, these properties aren't even supposed to be filled into forms because they won't be edited directly by the user anyway. I think it would be amazing if we could intentionally choose attributes we'd like to be filled in forms but, the default behavior would be to fetch all attributes just like how things are done right now.

**Comment by danharrin** [comment]
You have two options already - use Wireable objects, or use Filament's `mutateFormDataBeforeFill()` to customize or remove objects

**Reply by LogicSatinn** [reply]
I know of those options. But it would be better if you specified some attributes that were never fetched from the database in the first place. That way, you don't have to `mutateFormDataBeforeFill()`. 

In my case, I have two attributes, state and status uses State Machines, _Spatie's package_, I don't want these attributes hydrated by Livewire because they will never be updated by the user via forms anyway.

I truly hope I'm explaining this okay. I'm not good at explaining stuff.

**Reply by danharrin** [reply]
I don't quite see the difference between adding potentially 4 lines of code to tell Filament which columns not to select, and adding 4 lines of code to unset some properties in `mutateFormDataBeforeFill()`? I highly doubt there will be a performance difference

**Reply by LogicSatinn** [reply]
I know. I know. There won't be much of a performance difference or anything like that. 

I just think it's amazing if we had that control.

**Reply by danharrin** [reply]
Also, preventing data from loading onto the model does often have unintended consequences with Eloquent, if all the attributes are not hydrated. With getters and stuff that rely on those missing properties

**Reply by LogicSatinn** [reply]
Yeah, that's true.

**Reply by ghost** [reply]
> You have two options already - use Wireable objects, or use Filament's `mutateFormDataBeforeFill()` to customize or remove objects

How do and where do I use wireable objects? Do I implement Wireable on the model ie Post model or do I implement it on the resource?

I have a MoneyCast on a product model and implemented wireable on the model and it’s not working. I am new to Laravel so it could be me ;)

```
public function toLivewire(): array
    {
        return [
            'price' => Product::get('price'),
        ];
    }

    public static function fromLivewire($value): static
    {
        $price = $value['price'];

        return new static($price);
    }
```

![image](https://github.com/user-attachments/assets/c734abbb-92f8-4684-9a74-f47897af870b)
